File: java/998.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode insertIntoMaxTree(TreeNode root, int val) {
        if(root == null){
            return new TreeNode(val);
        }else{
            if(root.val < val){
                TreeNode node = new TreeNode(val);
                node.left = root;
                return node;
            }else{
                root.right = insertIntoMaxTree(root.right,val);
                return root;
            }
        }
    }
}

File: python/998.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def insertIntoMaxTree(self, root: TreeNode, val: int) -> TreeNode:
        if root == None:
            return TreeNode(val)
        else:
            if root.val < val:
                node = TreeNode(val)
                node.left = root
                return node
            else:
                root.right = self.insertIntoMaxTree(root.right,val)
                return root